src/main/java/org/sonarlint/intellij/config/project/AddEditExclusionDialog.java

```diff
@@ -0,0 +1,232 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+
+package org.sonarlint.intellij.config.project;
+
+import com.intellij.codeInsight.hint.HintManager;
+import com.intellij.codeInsight.hint.HintUtil;
+import com.intellij.openapi.fileChooser.FileChooserDescriptor;
+import com.intellij.openapi.fileChooser.FileChooserDescriptorFactory;
+import com.intellij.openapi.help.HelpManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.openapi.ui.TextFieldWithBrowseButton;
+import com.intellij.openapi.util.io.FileUtil;
+import com.intellij.openapi.vcs.changes.*;
+import com.intellij.ui.DocumentAdapter;
+import com.intellij.ui.HoverHyperlinkLabel;
+import com.intellij.ui.HyperlinkAdapter;
+import com.intellij.ui.HyperlinkLabel;
+import com.intellij.ui.awt.RelativePoint;
+import com.intellij.ui.components.JBLabel;
+import com.intellij.ui.components.JBTextField;
+import icons.SonarLintIcons;
+import java.awt.Cursor;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import javax.annotation.CheckForNull;
+import javax.swing.event.DocumentListener;
+import javax.swing.event.HyperlinkEvent;
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.NonNls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import javax.swing.event.DocumentEvent;
+import java.awt.event.ActionListener;
+import java.io.File;
+
+import static org.sonarlint.intellij.config.project.ExclusionItem.Type.DIRECTORY;
+import static org.sonarlint.intellij.config.project.ExclusionItem.Type.FILE;
+import static org.sonarlint.intellij.config.project.ExclusionItem.Type.GLOB;
+
+/**
+ * Based on IgnoreUnversionedDialog
+ */
+public class AddEditExclusionDialog extends DialogWrapper {
+  @Nullable
+  private final Project project;
+  private JPanel panel;
+
+  private JRadioButton fileRadioButton;
+  private JRadioButton directoryRadioButton;
+  private JRadioButton globRadioButton;
+
+  private TextFieldWithBrowseButton directoryTextField;
+  private JBTextField globTextField;
+  private TextFieldWithBrowseButton fileTextField;
+  private HyperlinkLabel patternHelp;
+  private JBLabel helpLabel;
+
+  public AddEditExclusionDialog(@Nullable final Project project) {
+    super(project, false);
+    this.project = project;
+    setTitle("Add SonarLint File Exclusion");
+    init();
+
+    FileChooserDescriptor fileChooser = new FileChooserDescriptor(true, false, false,
+      true, false, false);
+    if (project != null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2589)